inst/stan/estimate_truncation.stan

```diff
@@ -45,23 +48,32 @@ model {
   sigma ~ normal(0, 1) T[0,];
   // log density of truncated latest data vs that observed
   for (i in 1:(obs_sets - 1)) {
-    int start_t = t - obs_dist[i] - trunc_max;
-    for (j in 1:trunc_max) {
+    int end_t = t - obs_dist[i];
+    int start_t = max(0, end_t - trunc_max);
+    int trunc_t = end_t - start_t;
+    for (j in 1:trunc_t) {
       obs[start_t + j, i] ~ neg_binomial_2(trunc_obs[j, i], sqrt_phi);
     }
   }
 }
 generated quantities {
-  matrix[trunc_max, obs_sets] recon_obs;
+  matrix[trunc_max, obs_sets] recon_obs = rep_matrix(0, trunc_max, obs_sets);
   matrix[trunc_max, obs_sets - 1] gen_obs;
   // reconstruct all truncated datasets using posterior of the truncation distribution
   for (i in 1:obs_sets) {
     int end_t = t - obs_dist[i];
-    int start_t = end_t - trunc_max + 1;
-    recon_obs[, i] = truncate(to_vector(obs[start_t:end_t, i]), rev_cmf, 1);
+    int start_t = max(1, end_t - trunc_max + 1);
```

Comment:
in an ideal world it seems like these would be calculated in R as they are the same across the 3 blocks.

Reply:
moved to transformed data as an intermediate solution in https://github.com/epiforecasts/EpiNow2/pull/438/commits/63b6fd50bd915087292159c82f72b0e447016561